### PR TITLE
Edition summary page to GOVUK Design system

### PR DESCRIPTION
**Pull request by DilwoarH**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

[Move the Edition Summary page to the GOV.UK Design System](https://trello.com/c/voJfdq8h/685-move-edition-summary-page-to-the-govuk-design-system)

## Why

This is part of the transition work to move Whitehall to the GOVUK Design system.

## After

whitehall-admin dev gov uk_government_admin_detailed-guides_1331735(iPad Pro) (1)

## Before

whitehall-admin dev gov uk_government_admin_detailed-guides_1331735(iPad Pro)

### CODE REVIEW

**Comment by DilwoarH** [comment]
## Handover notes

### Things needed to get PR merged
- Border appears above Summary heading, if it cannot be removed, add a margin between them to space it out
- Tests are failing, mainly due to CSS selectors being different:
```
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:62
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:70
mtest test/functional/admin/publications_controller_test.rb:187
mtest test/functional/admin/publications_controller_test.rb:165
mtest test/functional/admin/publications_controller_test.rb:209
mtest test/functional/admin/publications_controller_test.rb:156
mtest test/functional/admin/publications_controller_test.rb:231
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:62
mtest test/functional/admin/consultations_controller_test.rb:133
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:70
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:70
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:62
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:70
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:62
mtest test/functional/admin/document_collections_controller_test.rb:15
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:70
mtest test/functional/admin/detailed_guides_controller_test.rb:31
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:62
mtest test/integration/page_title_test.rb:20
mtest test/functional/admin/news_articles_controller_test.rb:33
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:70
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:62
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:70
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:62
mtest test/functional/admin/fatality_notices_controller_test.rb:26
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:62
mtest test/support/admin_edition_controller_scheduled_publishing_test_helpers.rb:70
```

### Cards that need to be created
- Create a card in Trello to add back polling ajax for Broken Link Checker (make sure that the new version doesn't inject JS and works more securely)

File: app/views/admin/editions/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          { field: "Organisations", value: joined_list(@edition.organisations.map { |o| o.name }) },
+        ]
+      }
+    %>
+
+    <%=
+      list_items = [
+        link_to("Notes", admin_edition_editorial_remarks_path(@edition), class: "govuk-link"),
+        link_to("History", history_admin_edition_path(@edition), class: "govuk-link"),
+        *(link_to("Fact checking", admin_edition_fact_check_requests_path(@edition), class: "govuk-link") if @edition.can_be_fact_checked?),
```
Comment (davidgisbey) [inline app/views/admin/editions/show.html.erb]: I don't think these 3 links should render anymore now we will be readding tabs in.

File: app/views/admin/editions/show/_sidebar_link_checker.html.erb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    description: capture do %>
+      <% report.links.sort_by(&:status).group_by(&:status).each do |status, links| %>
+        <% next unless %w(broken caution).include? status %>
+
+        <p class="govuk-body"><%= t "broken_links.#{status}.subheading" %></p>
+
+        <ul class="govuk-list">
+          <% links.each do |link| %>
+            <li>
+              <%= link_to link.uri.truncate(50), link.uri, title: link.uri, class: 'govuk-link' %>
```
Comment (davidgisbey) [inline app/views/admin/editions/show/_sidebar_link_checker.html.erb:28]: Are we happy with the link moving out of the inset component? 

image

Comment (dnkrj) [reply app/views/admin/editions/show/_sidebar_link_checker.html.erb:28]: I think that this has already been fixed
https://github.com/alphagov/whitehall/pull/7083/commits/3da0aac35bd0776b5df18277ce4cb484398751fb#diff-1c67913b41e03a4fb87298119742d00de1f3fa686f195d9a793be6c41de7a869R57

**Comment by davidgisbey** [comment]
I've had a look through and it's behaving as expected. 

Are we happy with the test coverage? There's a lot going on in these views and as far as i can tell we're almost exclusively relying on feature specs. 

It feels like a lot of this could be refactored down the line into View Components and unit tested if we want to improve coverage.

I know we've not lost any test coverage though so maybe it's fine to merge and come back to later?

File: app/views/admin/editions/show/_main.html.erb [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+  <% end %>
+  <% if @edition.translatable? %>
+    <section class="app-view-edition-summary__section">
+      <%= render "govuk_publishing_components/components/heading", {
+        text: "Translations",
+        heading_level: 2,
+        font_size: "l",
+        margin_bottom: 3,
+      } %>
+      <% if @edition.editable? and @edition.missing_translations.any? %>
```
Comment (davidgisbey) [inline app/views/admin/editions/show/_main.html.erb:203]: Nit - I know this was the behaviour before, but i think we should use && unless theres a good reason not to throughout the codebase.

There's actually subtly different behaviour too which isn't obvious at a first glance https://stackoverflow.com/questions/1426826/difference-between-and-and-in-ruby

I know it doesn't directly affect this, bit i still think its better to just stick with `&&` unless there's a good reason not to.

File: app/views/admin/editions/show/_main.html.erb [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+      <% if @edition.non_english_translations.any? %>
+        <%= render "govuk_publishing_components/components/table", {
+          head: [
+            {
+              text: "Locale"
+            },
+            {
+              text: "Title"
+            },
+            (@edition.editable? or @edition.can_delete?) ? {
```
Comment (davidgisbey) [inline app/views/admin/editions/show/_main.html.erb:218]: Same as the above but with `||`

File: app/views/admin/editions/show/_sidebar_actions.html.erb [hunk 5]
```diff
@@ -0,0 +1,128 @@
+<%
+  items = []
+
+  if @edition.is_latest_edition? and @edition.published?
```
Comment (davidgisbey) [inline app/views/admin/editions/show/_sidebar_actions.html.erb:4]: Same as https://github.com/alphagov/whitehall/pull/7083/files#r1038020320

File: app/views/admin/editions/show/_sidebar_link_checker.html.erb (hunk 2 above)
Comment (davidgisbey) [reply app/views/admin/editions/show/_sidebar_link_checker.html.erb:28]: Oh okay i think the class needs to be added the broken links list then as its not being applied at the moment

**Comment by DilwoarH** [comment]
Sections missing:
- Change Notes
- World taxonomy
- Tagged to this document

File: app/views/admin/editions/show/_main.html.erb (hunk 3 above)
Comment (dnkrj) [reply app/views/admin/editions/show/_main.html.erb:209]: Thanks!

File: app/views/admin/editions/show/_sidebar_link_checker.html.erb (hunk 2 above)
Comment (dnkrj) [reply app/views/admin/editions/show/_sidebar_link_checker.html.erb:28]: Ah, I dropped the class from the DOM at some point
Have added it back

**Comment by davidgisbey** [comment]
Great. Thanks @dnkrj. Sounds good to me 👍

**Comment by dnkrj** [comment]
Missing sections, failing tests and other comments have now been addressed.
It looks like related documents were removed in https://github.com/alphagov/whitehall/pull/2220

File: features/admin-broken-link-reporting.feature [hunk 6]
```diff
@@ -2,7 +2,7 @@ Feature: Checking a document for broken links
 
   Documents can be checked for the presence of broken links.
 
-  @javascript
+  @javascript @design-system-wip
   Scenario: checking a draft document for broken links
```
Comment (davidgisbey) [inline features/admin-broken-link-reporting.feature:6]: One question on this. I think we're adding a wip to this as the JS has bot been added to load the broken links checker without reloading the page.

Do we know if we're planning on doing this or just asking the user to reload the page? And if we are planning on doing it, is there a card for it?

**Review by davidgisbey — APPROVED** [review]
lgtm, just 1 question about the link checker

File: features/admin-broken-link-reporting.feature (hunk 6 above)
Comment (dnkrj) [reply features/admin-broken-link-reporting.feature:6]: Yeah, exactly!
Have added a card for it here:
https://trello.com/c/doeqw0lS/990-reintroduce-polling-ajax-for-broken-link-checker

**Pull request merged**